### PR TITLE
ENT-3897: Fixed TypeError that pops up sometimes while communicating with the EMSI API.

### PR DESCRIPTION
**Pull request by saleem-latif**
__Description:__
During development, I found that sometimes connecting to EMSI Client raises `TypeError`. This is caused by the `client` attribute on EMSI Client being `None`. 

__Error Message:__
```
E           AttributeError: 'NoneType' object has no attribute 'versions'
```

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -14,6 +14,10 @@ Change Log
 Unreleased
 --------------------
 
+[1.2.0] - 2020-12-24
+--------------------
+* Fixed TypeError that pops up sometimes while communicating with the EMSI API.
```
Comment (sameenfatima78) [inline CHANGELOG.rst:19]: can you please add the linespace under date in the first two entries... i forgot to add in mine and have already merged my PR

File: taxonomy/emsi_client.py [hunk 2]
```diff
@@ -41,19 +41,26 @@ def __init__(self, scope):
...
         Return the cache key, caching is done based on the endpoint name and the scope property.
         """
         return get_cache_key(endpoint='EMSI', scope=self.scope)
 
-    def get_oauth_access_token(self, client_id, client_secret, grant_type='client_credentials'):
+    def fetch_oauth_access_token(self, client_id, client_secret, grant_type='client_credentials'):
         """
-        Return the access token if it is cached otherwise hit the endpoint and get a new access token.
+        Fetch a new access token from EMSI API.
+
```
Comment (sameenfatima78) [inline taxonomy/emsi_client.py:62]: this extra line can be removed and turned into one statement

**Review by sameenfatima78 — APPROVED** [review]
Looks good.. just two minor suggestions.

File: CHANGELOG.rst (hunk 1 above)
Comment (saleem-latif) [reply CHANGELOG.rst:20]: Done

File: taxonomy/emsi_client.py (hunk 2 above)
Comment (saleem-latif) [reply taxonomy/emsi_client.py]: Done

**Review by HammadAhmadWaqas — APPROVED** [review]

**Pull request merged**